From a01bc30a78a392681ec96e4d4d041c5c5227a6c5 Mon Sep 17 00:00:00 2001
From: Josh Wu <josh.wu@atmel.com>
Date: Tue, 26 Aug 2014 17:49:20 +0800
Subject: [PATCH] ARM: at91sam9rlek: add mmc environment configuration

This patch enable the mmc config, which will save the environment in a file: uboot.env

Signed-off-by: Josh Wu <josh.wu@atmel.com>
---
 boards.cfg                     |  1 +
 include/configs/at91sam9rlek.h | 18 +++++++++++++++++-
 2 files changed, 18 insertions(+), 1 deletion(-)

diff --git a/boards.cfg b/boards.cfg
index 1ba2081..48ec241 100644
--- a/boards.cfg
+++ b/boards.cfg
@@ -111,6 +111,7 @@ Active  arm         arm926ejs      at91        atmel           at91sam9m10g45ek
 Active  arm         arm926ejs      at91        atmel           at91sam9n12ek       at91sam9n12ek_mmc                     at91sam9n12ek:AT91SAM9N12,SYS_USE_MMC                                                                                             Josh Wu <josh.wu@atmel.com>
 Active  arm         arm926ejs      at91        atmel           at91sam9n12ek       at91sam9n12ek_nandflash               at91sam9n12ek:AT91SAM9N12,SYS_USE_NANDFLASH                                                                                       Josh Wu <josh.wu@atmel.com>
 Active  arm         arm926ejs      at91        atmel           at91sam9n12ek       at91sam9n12ek_spiflash                at91sam9n12ek:AT91SAM9N12,SYS_USE_SPIFLASH                                                                                        Josh Wu <josh.wu@atmel.com>
+Active  arm         arm926ejs      at91        atmel           at91sam9rlek        at91sam9rlek_mmc                      at91sam9rlek:AT91SAM9RL,SYS_USE_MMC                                                                                               Stelian Pop <stelian@popies.net>
 Active  arm         arm926ejs      at91        atmel           at91sam9rlek        at91sam9rlek_dataflash                at91sam9rlek:AT91SAM9RL,SYS_USE_DATAFLASH                                                                                         Stelian Pop <stelian@popies.net>
 Active  arm         arm926ejs      at91        atmel           at91sam9rlek        at91sam9rlek_nandflash                at91sam9rlek:AT91SAM9RL,SYS_USE_NANDFLASH                                                                                         Stelian Pop <stelian@popies.net>
 Active  arm         arm926ejs      at91        atmel           at91sam9x5ek        at91sam9x5ek_dataflash                at91sam9x5ek:AT91SAM9X5,SYS_USE_DATAFLASH                                                                                         Bo Shen <voice.shen@atmel.com>
diff --git a/include/configs/at91sam9rlek.h b/include/configs/at91sam9rlek.h
index e5c9310..1f493fc 100644
--- a/include/configs/at91sam9rlek.h
+++ b/include/configs/at91sam9rlek.h
@@ -157,7 +157,7 @@
 				"mtdparts=atmel_nand:-(root) "\
 				"rw rootfstype=jffs2"
 
-#else /* CONFIG_SYS_USE_NANDFLASH */
+#elif CONFIG_SYS_USE_NANDFLASH
 
 /* bootstrap + u-boot + env + linux in nandflash */
 #define CONFIG_ENV_IS_IN_NAND		1
@@ -170,6 +170,22 @@
 				"mtdparts=atmel_nand:128k(bootstrap)ro,256k(uboot)ro,128k(env1)ro,128k(env2)ro,2M(linux),-(root) " \
 				"rw rootfstype=jffs2"
 
+#else /* CONFIG_SYS_USE_MMC */
+
+/* bootstrap + u-boot + env + linux in mmc */
+#define CONFIG_ENV_IS_IN_FAT
+#define CONFIG_FAT_WRITE
+#define FAT_ENV_INTERFACE	"mmc"
+#define FAT_ENV_FILE		"uboot.env"
+#define FAT_ENV_DEVICE_AND_PART	"0"
+#define CONFIG_ENV_SIZE		0x4000
+#define CONFIG_BOOTCOMMAND	"fatload mmc 0:1 0x21000000 at91sam9rlek.dtb; " \
+				"fatload mmc 0:1 0x22000000 zImage; " \
+				"bootz 0x22000000 - 0x21000000"
+#define CONFIG_BOOTARGS		"console=ttyS0,115200 " \
+				"mtdparts=atmel_nand:" \
+				"8M(bootstrap/uboot/kernel)ro,-(rootfs) " \
+				"root=/dev/mmcblk0p2 rw rootwait"
 #endif
 
 #define CONFIG_SYS_PROMPT		"U-Boot> "
-- 
1.9.1

